src/components/UserSelector.tsx

```diff
@@ -2,15 +1,56 @@
+import classNames from 'classnames';
+import React, { useRef, useState } from 'react';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+interface Props {
+  users: User[];
+  userId: number | null;
+  setUserId: React.Dispatch<React.SetStateAction<number | null>>;
+  setSelectedpost: React.Dispatch<React.SetStateAction<Post | null>>;
+}
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  userId,
+  setUserId,
+  setSelectedpost,
```

Comment:
The prop `setSelectedpost` should be renamed to `setSelectedPost` to maintain consistency with the naming convention used for state variables and props.